myning/tui/chapter/__init__.py

```diff
@@ -166,21 +167,20 @@ def clear(self):
         self.pick(PickArgs(message="", options=[]))
 
 
-def get_labels_and_hotkeys(options: list[OptionLabel]):
+def get_labels_and_hotkeys(options: list[Option]) -> tuple[list[list[str | Text]], dict[str, int]]:
```

Comment:
this return type doesnt look fun, can we leave it implicit?

Reply:
How is implicit better? I think if the return type looks gross written explicitly, that's a code smell, not a typing problem.

Reply:
I think about the value that a type hint brings when considering whether it's beneficial to add or keep them, for example, typing parameters gives us autocomplete in the body of the function, so it's always great to have types on parameters. Return types would be used when we work with the result of calling the function, but those types are already available implicitly and we have autocompletion, so the return type essentially adds nothing except for when you look at this one line of code and want to know what it returns.

In this one case, the return types of this function are defined in the next two lines so there's not a huge benefit to wrapping the two things in a tuple and creating this fun-looking return type.

Some examples of good reasons to add type hints for return types are when then inferred return type is a subset of the actual type you want to return. One example is with overloads:
```python
# myning/objects/inventory.py

class Inventory:
    ...

    @overload
    def get_slot(self, slot: Literal[ItemType.PLANT]) -> list[Plant]:
        ...

    @overload
    def get_slot(self, slot: ItemType) -> list[Item]:
        ...

    def get_slot(self, slot) -> list[Item] | list[Plant]:
        return self._items.get(slot, [])
```

Without this, when we're trying to get the list of `Plant` from the inventory, it would be typed as a list of `Item` and we wouldn't have access to attributes on `Plant`.

I'm not saying with absolutely have to remove this; I'm okay if we feel it's beneficial, but I'm just pointing out that we could have less code to read and maintain by relying on implicit typing when it's straightforward.

By the way, the `inlay hints python` setting in vscode might be handy if you like to use it

Reply:
After working in go for long enough and recently switching to Python, I've realized that typing saves a ton of headache down the road. Even if it only adds a benefit for one row, I think that's a win, especially if the alternative is having @jeremyd85  go through and remove the explicit typing he's already written.

Reply:
leaving it the way it is now is fine 👍